src/main/java/org/apache/sysds/runtime/util/DataConverter.java

```diff
@@ -456,6 +456,23 @@ public static MatrixBlock convertToMatrixBlock( double[][] data ) {
 		return mb;
 	}
 
+
+	/**
+	 * Converts an Integer matrix to an MatrixBlock (FP32)
+	 * 
+	 * @param data Int matrix input that is converted to double MatrixBlock
+	 * @return The matrixBlock constructed.
+	 */
+	public static MatrixBlock convertToMatrixBlock(int[][] data){
+		double[][] input = new double[data.length][];
+		int index = 0;
+		for(int[] x: data){
+			input[index] = Arrays.stream(x).asDoubleStream().toArray();
```

Comment:
If you want to use streams here, use `Arrays.setAll(input[index], i -> data[index][i])`